File: eMaestroD.NgUI/src/app/Invoices/Models/invoice.ts
export interface Invoice {
  invoiceID?: number;
  invoiceDetailID? :number;
  invoiceDate?: Date;
  bookerID?:number;
  salesmanID?:number;
  invoiceVoucherNo?:any;
  invoiceType?:any;
  txtypeID?:number;
  CustomerOrVendorID?:number;
  customerOrVendorName?:string;
  fiscalYear? : number;
  comID?:number;
  locID?:number;
  grossTotal?: number;
  totalDiscount?: number;
  totalTax?: number;
  totalRebate?: number;
  totalExtraTax?: number;
  totalAdvanceExtraTax?: number;
  totalExtraDiscount?: number;
  netTotal?: number;
  notes?:string;
  convertedInvoiceNo?:any;
  isPaymented? : boolean;
  isApproved?:boolean;
  transactionStatus? : string;
  totalRemainingPayment?:number;
  Products?: InvoiceProduct[];
}


export interface InvoiceProduct {
  prodInvoiceID? :number;
  prodID?: number;
  prodBCID?: number;
  prodCode?: any;
  prodName?: any;
  descr?: any;
  unitQty?: number;
  qty?: number;
  bounsQty?: number;
  notes?:any;
  batchNo?:any;
  expiry?:any;
  purchRate?: number;
  sellRate?: number;
  discountPercent?: number;
  discountAmount?: number;
  extraDiscountPercent?: number;
  extraDiscountAmount?: number;
  rebatePercent?:number;
  rebateAmount?:number;
  grossValue?: number;
  netAmount?: number;
  mrp?:number;
  sellingPrice?:number;
  lastCost?:number;
  unit?:string;
  ProductTaxes? : InvoiceProductTax[];
}

export interface InvoiceProductTax
{
    taxDetailID?:number;
    taxAcctNo?:string;
    taxPercent?:number;
    taxAmount?:number;
}

File: Models/VMModels/Invoice.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eMaestroD.Models.VMModels
{
    public class Invoice
    {
        public int? invoiceID { get; set; }
        public DateTime? invoiceDate { get; set; }
        public string? invoiceVoucherNo { get; set; }
        public string? invoiceType { get; set; }
        public int? fiscalYear { get; set; }
        public int? txTypeID { get; set; }
        public int? CustomerOrVendorID { get; set; }
        public string? customerOrVendorName { get; set; }
        public int? comID { get; set; }
        public int? locID { get; set; }
        public decimal? grossTotal { get; set; }
        public decimal? totalDiscount { get; set; }
        public decimal? totalTax { get; set; }
        public decimal? totalRebate { get; set; }
        public decimal? totalExtraTax { get; set; }
        public decimal? totalAdvanceExtraTax { get; set; }
        public decimal? totalExtraDiscount { get; set; }
        public decimal? netTotal { get; set; }
        public string? notes { get; set; }
        public string? convertedInvoiceNo { get; set; }
        public bool? isPaymented { get; set; }
        // List of products in the invoice
        public List<InvoiceProduct> Products { get; set; }
    }
}
